### PR TITLE
fix: time restriction upper bound conversion issues

### PR DESCRIPTION
**Pull request by richardtreier**
closes https://github.com/sovity/edc-ui/issues/815

### CODE REVIEW

File: src/app/core/services/api/fake-backend/connector-fake-impl/data/test-policies.ts [hunk 1]
```diff
@@ -26,12 +26,12 @@ export namespace TestPolicies {
       constraint(
         policyLeftExpressions.policyEvaluationTime,
         'GEQ',
-        '2020-12-31T23:00:00.000Z',
+        '2020-11-30T23:00:00.000Z',
       ),
       constraint(
         policyLeftExpressions.policyEvaluationTime,
         'LT',
```
Comment (kamilczaja) [inline src/app/core/services/api/fake-backend/connector-fake-impl/data/test-policies.ts:33]: Shouldn't this be `LEQ`?

File: src/app/core/utils/date-utils.ts [hunk 2]
```diff
@@ -4,8 +4,67 @@
  * Can be used to ensure dates are displayed identically across different timezones when stringified in JSON payloads.
  * @param date date to convert
  */
+import {addDays, subDays} from 'date-fns';
 import {format} from 'date-fns-tz';
 
 export function toGmtZeroHourDate(date: Date): Date {
   return new Date(format(date, 'yyyy-MM-dd'));
 }
```
Comment (kamilczaja) [inline src/app/core/utils/date-utils.ts:12]: Is this doc relating to `toGmtZeroHourDate`? Even if not, why is this above the imports?

Comment (richardtreier) [reply src/app/core/utils/date-utils.ts:12]: fixed

File: src/app/core/services/api/fake-backend/connector-fake-impl/data/test-policies.ts (hunk 1 above)
Comment (richardtreier) [reply src/app/core/services/api/fake-backend/connector-fake-impl/data/test-policies.ts:33]: no, check the month segment. UTC is minus one day at 22:00 or 23:00 for our midnight depending on summer time / winter time

**Review by kamilczaja — APPROVED** [review]
LGTM

**Pull request merged**